### PR TITLE
DM-44487: Fix sphgeom PyPi upload

### PR DESCRIPTION
**Pull request by mwittgen**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

**Comment by timj** [comment]
What's the motivation for dropping 3.8 and 3.9? I know we don't care that much but did they break somehow?

File: pyproject.toml [hunk 1]
```diff
@@ -22,9 +22,9 @@ classifiers = [
     "Operating System :: POSIX :: Linux",
     "Operating System :: MacOS :: MacOS X",
     "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.8",
-    "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
```
Comment (timj) [inline pyproject.toml:27]: Might want to add:

```toml
requires-python = ">=3.10.0"
```

in the toml section as well.

**Comment by mwittgen** [comment]
Build time and we don't  support it any more on other packages.

**Review by timj — APPROVED** [review]
Thanks. My main question is the python version change.

**Comment by timj** [comment]
> Build time and we don't support it any more on other packages.

Okay. I don't think anyone is going to care.

**Comment by fritzm** [comment]
Well, you might reasonably think that, but we might have Qserv bustage...   We're still on 3.8 for Python integration, regrettably, where Nate Pease left it...

**Comment by mwittgen** [comment]
OK, putting 3.8 back.

**Comment by fritzm** [comment]
Oh, but nevermind -- recalling we pull in `sphgeom` as a submodule for precisely these reasons :-)  So no problem.

**Comment by mwittgen** [comment]
@fritzm Let's leave 3.8/9 in for now. Better do topical PRs and do this later.

**Review by fritzm — APPROVED** [review]
LGTM, thanks!

**Comment by timj** [comment]
> Oh, but nevermind -- recalling we pull in sphgeom as a submodule for precisely these reasons :-) So no problem.

@fritzm it would be a problem, because no-one would know when we broke 3.8 if we were never testing it in the github action. Note that I was completely assuming that at any time I could go straight to 3.11 or whatever the current pipeline standard was. I've only been keeping it running with 3.8 because nothing was breaking with 3.8.

**Comment by fritzm** [comment]
Right.  What I meant to say was "no problem for Qserv" (we'll continue to build successfully against our effectively "pinned" github SHA in the Qserv sphgeom submodule just fine until we actively choose to port forward to latest.)

**Pull request merged**